public/js/app/initializers/PubSub.js

```diff
@@ -3,266 +3,271 @@ define(["config",
+          var comment = this.get('main_store').getById('comment', data.comments.id)
+          comment.set('isRealtime', true)
+
+          post.get('comments').pushObject(comment)
+        }
+      } else {
+        this.get('main_store').find('post', data.comments.postId)
+          .then(function(post) {
+            that.currentController().get('model.posts').unshiftObject(post)
+          })
+      }
+    },
+
+    updateComment: function(data) {
+      var commentId = data.comments.id
+
+      if (this.get('main_store').recordIsLoaded('comment', commentId)) {
+        var comment = this.get('main_store').getById('comment', commentId)
+        comment.set('body', data.comments.body)
+      }
+    },
+
+    destroyComment: function(data) {
+      var comment = this.get('main_store').getById('comment', data.commentId)
+
+      if (!comment)
+        return
+
+      if (!comment.get('isDeleted')) {
+        this.get('main_store').unloadRecord(comment)
+
+        var post = this.get('main_store').getById('post', data.postId)
+        comment = post.get('comments').findProperty('id', comment.get('id'))
+        post.get('comments').removeObject(comment)
+      }
+    },
+
+    newLike: function(data) {
+      if (!this.isFirstPage())
+        return
+
+      var that = this
+      var userId = data.users.id
+      var currentUser = this.currentController().get('session.currentUser')
+      var banIds = []
+      if (currentUser)
+        banIds = currentUser.get('banIds')
+
+      if (banIds.indexOf(userId) >= 0)
+        return
+
+
+      if (!this.get('main_store').recordIsLoaded('user', userId)) {
+        this.get('main_store').pushPayload('user', data)
+      }
+
+      var post = this.get('main_store').getById('post', data.meta.postId)
+
+      if (post) {
+        var user = this.get('main_store').getById('user', userId)
+        post.get('likes').addObject(user)
+      } else {
+        this.get('main_store').find('post', data.meta.postId)
+          .then(function(post) {
+            that.currentController().get('model.posts').unshiftObject(post)
+          })
+      }
+    },
+
+    removeLike: function(data) {
+      var post = this.get('main_store').getById('post', data.meta.postId)
+
+      if (post) {
+        var user = post.get('likes').findProperty('id', data.meta.userId)
+        post.get('likes').removeObject(user)
+      }
+    }
+  })
 
-        removeLike: function(data) {
-          var post = this.store.getById('post', data.meta.postId)
+  Ember.Application.initializer({
+    name: 'pubsub',
+    after: "store",
 
-          if (post) {
-            var user = post.get('likes').findProperty('id', data.meta.userId)
-            post.get('likes').removeObject(user)
-          }
-        }
-      })
+    initialize: function(container, application) {
+      App.PubSubService = PubSubServiceClass.create({
+        main_router: container.lookup('router:main'),
+        main_store: container.lookup('store:main')
+      });
```

Comment:
Let's inject PubSub into Router and Controller as well:
```
application.register('user:pubsub', App.PubSubService, { instantiate: false, singleton: true })
application.inject('route', 'pubsub', 'user:pubsub')
application.inject('controller', 'pubsub', 'user:pubsub')
```
This will address my recent comment about using App.PubSubService directly.